packages/vscode-extension/src/builders/buildIOS.ts

```diff
@@ -228,20 +261,30 @@ async function buildLocal(
 
   Logger.debug(`Xcode build will use "${scheme}" scheme`);
 
+  const { simulatorUdid, cleanupSimulator } = await getOrMakeSimulator(buildConfig);
+  if (cleanupSimulator) {
+    if (cancelToken.cancelled) {
+      cleanupSimulator();
+    } else {
+      cancelToken.onCancel(cleanupSimulator);
+    }
+  }
```

Comment:
I think It'd be best to just wrap the whole of this function's body after the temp simulator is created in a `try`/`finally { cleanup(); }`, that way there shouldn't be any risk of it failing us.

Right now, if `await buildProcess` [at line 292](https://github.com/software-mansion/radon-ide/pull/1461/files#diff-c1465ea393e4601b8aa787aa23cbffe33457e2fe8e62608a248b33ba0a39eaafL249) fails, it looks to me that the simulator is never cleaned up.